### ISSUE TITLE
Type definitions are incompatible with Vue3

### ISSUE THREAD

**Issue by j2L4e** [highcharts/highcharts-vue#209]
highcharts-vue.d.ts imports several types that Vue3 no longer exports.

See https://github.com/vuejs/core/issues/1370

```
node_modules/highcharts-vue/types/highcharts-vue.d.ts:1:23 - error TS2305: Module '"vue"' has no exported member 'PropOptions'.

1 import _Vue, { VNode, PropOptions, WatchOptionsWithHandler, WatchHandler, CreateElement } from "vue";
                        ~~~~~~~~~~~

node_modules/highcharts-vue/types/highcharts-vue.d.ts:1:36 - error TS2305: Module '"vue"' has no exported member 'WatchOptionsWithHandler'.

1 import _Vue, { VNode, PropOptions, WatchOptionsWithHandler, WatchHandler, CreateElement } from "vue";
                                     ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/highcharts-vue/types/highcharts-vue.d.ts:1:61 - error TS2305: Module '"vue"' has no exported member 'WatchHandler'.

1 import _Vue, { VNode, PropOptions, WatchOptionsWithHandler, WatchHandler, CreateElement } from "vue";
                                                              ~~~~~~~~~~~~

node_modules/highcharts-vue/types/highcharts-vue.d.ts:1:75 - error TS2305: Module '"vue"' has no exported member 'CreateElement'.

1 import _Vue, { VNode, PropOptions, WatchOptionsWithHandler, WatchHandler, CreateElement } from "vue";
                                                                            ~~~~~~~~~~~~~

node_modules/highcharts-vue/types/highcharts-vue.d.ts:7:42 - error TS2709: Cannot use namespace '_Vue' as a type.

7     interface ComponentOptions<V extends _Vue> {
                                           ~~~~

node_modules/highcharts-vue/types/highcharts-vue.d.ts:33:28 - error TS2507: Type 'typeof import("/redacted/node_modules/vue/dist/vue")' is not a constructor function type.

33 export class Chart extends _Vue {
                              ~~~~

```

**Comment by Denyllon** [comment]
Hi @j2L4e 

Thank you for reporting the problem.
Indeed, it looks like the bug which we should resolve immediately, but definitely need to have some time investigate it.
Meanwhile, I'm marking this issue with **Bug** label.

Kind regards!

**Comment by pwang2** [comment]
@Denyllon, just want to kindly ask if any update on this?

**Comment by Denyllon** [comment]
Unfortunately it's not resolved yet. I think in this case we need to figure out what we should do with our existing `d.ts` file, as the latest Vue version has completely different exported members. The most important thing here is to keep this wrapper compatible with both Vue 2 and 3 versions, what could be a bit problematic.

**Comment by pwang2** [comment]
It makes more sense to skip the default loading of d.ts and allow the dev tools to decide which one to load.

**Comment by IceBlizz6** [comment]
Do we have an update on this?
Would be great to get proper type support for this library.

**Comment by Denyllon** [comment]
Unfortunately, it's not fixed yet. We'll schedule the work on type definitions soon.

**Comment by georgebohnisch** [comment]
Updates?

**Comment by pwang2** [comment]
before any fix available. I was adding` "postinstall": "rimraf node_modules/highcharts-vue/types"` to package.json scripts.   and using Hightcharts's 150K lines of typing file. we might just need types for the chartOptions.

**Comment by jaclynt** [comment]
Any updates? Still encountering this issue.

**Comment by jakubSzuminski** [comment]
I've prioritized this issue and put it in our backlog. I will keep you posted.

**Comment by rcheung9** [comment]
Still an issue.  PR not approved.

**Comment by jakubSzuminski** [comment]
You're right @rcheung9. I'm sorry, but I did not manage to resolve this yet as there have been prioritized Highcharts Export Server issues that I had to look into.

I will try to merge a fix and publish a new version by the end of January.

In the meantime, you can override the default types with these types that I've prepared:
```TypeScript
// Types for Vue3
import { DefineComponent, App } from 'vue';
import * as Highcharts from 'highcharts';

export type ChartUpdateArgs = [boolean, boolean, Highcharts.AnimationOptionsObject];

declare module '@vue/runtime-core' {
    export interface ComponentCustomProperties {
        highcharts?: typeof Highcharts;
    }
}

export interface ChartComponentOptions {
    constructorType?: string;
    options?: Highcharts.Options;
    callback?: Highcharts.ChartCallbackFunction;
    updateArgs?: ChartUpdateArgs;
    highcharts?: typeof Highcharts;
    deepCopyOnUpdate?: boolean;
}

export interface ChartProps {
    constructorType?: string;
    options: Highcharts.Options;
    updateArgs?: ChartUpdateArgs;
    callback?: Highcharts.ChartCallbackFunction;
}

export const Chart: DefineComponent<ChartProps>;

export default function install(app: App, options?: ChartComponentOptions): void;
```

which works well with the latest version of Vue.

**Comment by Pedrocanoas** [comment]
this error is related?

![image](https://github.com/highcharts/highcharts-vue/assets/51349076/65182413-7dcd-4112-b555-698b46ab7797)

```
  "dependencies": {
    "axios": "1.6.7",
    "highcharts": "^11.3.0",
    "highcharts-vue": "^1.4.3",
    "moment": "^2.30.1",
    "vue": "^3.4.15",
    "vue-router": "^4.0.11"
  },
  ```

**Comment by Pedrocanoas** [comment]
> this error is related?
> 
> ![image](https://private-user-images.githubusercontent.com/51349076/303012434-65182413-7dcd-4112-b555-698b46ab7797.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDczODkzOTUsIm5iZiI6MTcwNzM4OTA5NSwicGF0aCI6Ii81MTM0OTA3Ni8zMDMwMTI0MzQtNjUxODI0MTMtN2RjZC00MTEyLWI1NTUtNjk4YjQ2YWI3Nzk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA4VDEwNDQ1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1MjYwMTQxNmI1NjAxOGUwNmRhYmQxMjM3OWYyNjA3YzQzNzI3MTVmYWU0OTJmMDNhYWRhYjBhNzE3MTYyYmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.75LTHuX4Php3FgFOMGUgvjeedJ4yZZosjAF0KRrfpu8)
> 
> ```
>   "dependencies": {
>     "axios": "1.6.7",
>     "highcharts": "^11.3.0",
>     "highcharts-vue": "^1.4.3",
>     "moment": "^2.30.1",
>     "vue": "^3.4.15",
>     "vue-router": "^4.0.11"
>   },
> ```

I don't know why, but I fixed the previous error setting the type

![image](https://github.com/highcharts/highcharts-vue/assets/51349076/3427c115-2224-4321-9877-e1df6ca2810d)

**Comment by jakubSzuminski** [comment]
Yes @Pedrocanoas, this comment might be related if you're using TypeScript. I have created a new branch for a Vue3 full support release: https://github.com/highcharts/highcharts-vue/pull/258 which I schedule for this week.

**Comment by jakubSzuminski** [comment]
@Pedrocanoas did not manage to release it yet, currently on the testing phase. Feel free to install the beta version from the GitHub branch like this:

```JSON
"dependencies": {
    "vue": "^3.4.15",
    "highcharts": "10.3.3",
    "highcharts-vue": "https://github.com/highcharts/highcharts-vue.git#vue3/types"
  },
```

Everything should work as expected, but please try it out and let me know about your experience.

**Comment by hexagon06** [comment]
@jakubSzuminski The feature branch version works like a charm for me!
Any idea when the new version will be out so I can update the dependency version?

**Comment by jakubSzuminski** [comment]
We've just released v2.0.0. Closing via https://github.com/highcharts/highcharts-vue/pull/258.

Please let me know if there's anything that does not work as expected. We'll try to fix it asap.
